src/v3/src/transformer/button/transformIDPButtons.ts

```diff
@@ -10,93 +10,58 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { NextStep } from '@okta/okta-auth-js';
-
 import { IDX_STEP } from '../../constants';
-import SmartCardIconSvg from '../../img/smartCardButtonIcon.svg';
 import {
-  ButtonElement,
-  ButtonType,
   DividerElement,
-  IWidgetContext,
   TransformStepFnWithOptions,
 } from '../../types';
-import { loc } from '../../util';
+import { getFastPassButtonElement, getIdpButtonElements, loc } from '../../util';
 
-// TODO: OKTA-504638 Implement CUSTOM IDP Buttons here
 export const transformIDPButtons: TransformStepFnWithOptions = ({
   transaction,
   widgetProps,
 }) => (
   formbag,
 ) => {
   const { neededToProceed: remediations } = transaction;
-  const containsPivIDP = remediations.some(
-    (remediation) => remediation.name === IDX_STEP.PIV_IDP,
-  );
   const containsIdentifyStep = remediations.some(
     (remediation) => remediation.name === IDX_STEP.IDENTIFY,
   );
-  // To display button must be on identify page and contain PIV remediation
-  if (!(containsPivIDP && containsIdentifyStep) ) {
+  const containsEnrollProfileStep = remediations.some(
+    (remediation) => remediation.name === IDX_STEP.ENROLL_PROFILE,
+  );
+
+  // must be on identify or enroll profile page to display Idp buttons
+  if (!(containsIdentifyStep || containsEnrollProfileStep)) {
     return formbag;
   }
 
-  const { piv, idpDisplay } = widgetProps;
-  const isBottomDivider = typeof idpDisplay === 'undefined'
-    ? true
-    : idpDisplay.toUpperCase() === 'PRIMARY';
```

Comment:
We may want to keep this since it is a supported feature in Gen 2: https://github.com/okta/okta-signin-widget#idpdisplay

Reply:
based on this comment from Nara I thought we were going to remove this feature: https://oktainc.atlassian.net/browse/OKTA-602160?focusedCommentId=3581371

Reply:
ah I see we can still keep this feature but just set the IDP buttons to the top by default,  but still allow the user the set the idpDisplay to `SECONDARY` if they want to display them at the bottom.

Reply:
Yep exactly, if I am not mistaken I believe that is how we did it for Gen 3 with PIV (displayed it at the top by default instead of bottom unless they configured it to be displayed at the bottom).